platform-common-date/src/main/java/com/softicar/platform/common/date/DayParser.java

```diff
@@ -32,15 +32,38 @@ public DayParser(String text) {
 		this.text = text;
 	}
 
-	public Optional<Day> parse() {
+	/**
+	 * Parses the given text into a {@link Day}.
+	 *
+	 * @return the parsed {@link Day} object (never <i>null</i>)
+	 * @throws IllegalDateSpecificationException
+	 *             if the given text could not be parsed
+	 */
+	public Day parseOrThrowIfInvalid() {
 
 		try {
 			return FORMATS//
 				.stream()
 				.filter(format -> format.matches(text))
 				.findFirst()
-				.map(format -> format.parse(text));
+				.map(format -> format.parse(text))
+				.orElseThrow(() -> new IllegalDateSpecificationException(text));
 		} catch (NumberFormatException exception) {
+			throw new IllegalDateSpecificationException(exception, text);
+		}
+	}
+
+	/**
+	 * Parses the given text into a {@link Day}.
+	 *
+	 * @return the parsed {@link Day} object as {@link Optional}; an empty
+	 *         {@link Optional} if parsing failed
```

Comment:
I suggest a direct link:
`...; if parsing fails, {@link Optional#empty()} is returned`

Reply:
did so